### PR TITLE
[DOP-19900] Add MSSQL API schema

### PR DESCRIPTION
**Pull request by IlyasDevelopment**
## Change Summary
- Added Pydantic schemas for MSSQL connection
- Added Pydantic schemas for MSSQL source parameters and target parameters
- Added unit tests for creating & updating MSSQL connection and target
- Updated README to include references to MSSQL 

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [ ] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [x] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/syncmaster/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: tests/test_unit/test_connections/connection_fixtures/group_connections_fixture.py [hunk 1]
```diff
@@ -17,13 +22,30 @@ async def group_connections(group_connection: MockConnection, session: AsyncSess
...
+                    "cluster": "cluster",
+                },
+            )
+        elif conn_type == ConnectionType.S3:
+            new_data.update(
+                {
+                    "bucket": "bucket",
+                },
+            )
+        elif conn_type in [ConnectionType.ORACLE, ConnectionType.CLICKHOUSE, ConnectionType.MSSQL]:
```
Comment (dolfinus) [inline tests/test_unit/test_connections/connection_fixtures/group_connections_fixture.py:42]: Postgres is missing

Comment (IlyasDevelopment) [reply tests/test_unit/test_connections/connection_fixtures/group_connections_fixture.py:48]: Actually, first group_connection fixture already created a connection with Postgres, but it still should be checked if we will change this fixture, added

Comment (dolfinus) [reply tests/test_unit/test_connections/connection_fixtures/group_connections_fixture.py:48]: Why some DB connections have `database_name` field, and others have `database`?

Comment (IlyasDevelopment) [reply tests/test_unit/test_connections/connection_fixtures/group_connections_fixture.py:48]: I think `database` is more appropriate, let's use this option

Comment (IlyasDevelopment) [reply tests/test_unit/test_connections/connection_fixtures/group_connections_fixture.py:48]: A lot of places need to be fixed for name congruence, should it be done in a separate PR?

Comment (dolfinus) [reply tests/test_unit/test_connections/connection_fixtures/group_connections_fixture.py:48]: > A lot of places need to be fixed for name congruence, should it be done in a separate PR?

Yep

Comment (dolfinus) [reply tests/test_unit/test_connections/connection_fixtures/group_connections_fixture.py:48]: For Oracle fields are `service_name` and `sid`. For other JDBC types they should have the same name, either `database_name` or `database`. IMHO `database_name` is more descriptive.

**Pull request merged**